.github/workflows/actions/diff/action.yml

```diff
@@ -0,0 +1,95 @@
+name: Build and diff
+
+outputs:
+  changes:
+    description: 'Returns true if build output has changes'
+    value: ${{ steps.diff.outputs.changes || 'true' }}
+
+runs:
+  using: composite
+
+  steps:
+    - name: Set up diff drivers
+      shell: bash
+      run: |
+        npm install -g js-beautify
+        git config diff.minjs.textconv "js-beautify --editorconfig --type js"
+        git config diff.mincss.textconv "js-beautify --editorconfig --type css"
+        git config user.name github-actions
+        git config user.email github-actions@github.com
+
+    - name: Clear diff cache
+      shell: bash
+      run: |
+        rm -Rf .cache/diff/dist
+        mkdir -p .cache/diff/dist
+
+    - name: Restore base build (from cache)
+      uses: actions/cache/restore@v3
+      id: cache-base
+      with:
+        path: dist/
+        key: build-${{ github.event.pull_request.base.sha }}
+
+    - name: Install and build base
+      if: ${{ steps.cache-base.outputs.cache-hit != 'true' }}
+      shell: bash
+      run: |
+        git checkout ${{ github.base_ref }}
+        npm install --ignore-scripts --no-save --silent
+        npm run build
+
+    - name: Save base build (to cache)
+      if: ${{ steps.cache-base.outputs.cache-hit != 'true' }}
+      uses: actions/cache/save@v3
+      with:
+        path: dist/
+        key: build-${{ github.event.pull_request.base.sha }}
+
+    - name: Commit base build for comparison
+      shell: bash
+      run: |
+        git checkout ${{ github.head_ref }}
+        git add --force dist/
+        git commit --allow-empty -m "Build output for '${{ github.base_ref }}'" --no-verify
+
+    - name: Restore head build (from cache)
+      uses: actions/cache/restore@v3
+      id: cache-head
+      with:
+        path: dist/
+        key: build-${{ github.event.pull_request.head.sha }}
+
+    - name: Install dependencies
+      if: ${{ steps.cache-head.outputs.cache-hit == 'true' }}
+      shell: bash
+      run: npm install
```

Comment:
Why do we need this `npm install`?  We've already got the build (from the cache) and we don't run anything after this point that uses the dependencies, unless I've missed something.